appinventor/appengine/src/com/google/appinventor/client/wizards/ProjectUploadWizard.java

```diff
@@ -58,48 +58,134 @@ public void execute() {
               substring(Math.max(filename.lastIndexOf('/'), filename.lastIndexOf('\\')) + 1);
 
           // Make sure the project name is legal and unique.
-          if (!TextValidators.checkNewProjectName(filename)) {
-            return;
-          }
+          if (!TextValidators.checkNewProjectName(filename, true)) {
+         
+            String suggestedName = "";
+            String title;
+            filename = filename.replace(" ", "_").replaceAll("[^a-zA-Z0-9_]", "");
+
+            if (TextValidators.isTitleReserved()) {
+              title = MESSAGES.reservedTitleFormatError() + " : " + filename;
```

Comment:
I would recommend making the titles take arguments so that when translators translate them into other language the position of the filename can be moved if it makes sense to do so.